spring-integration-scripting/src/main/java/org/springframework/integration/scripting/dsl/DslScriptExecutingMessageProcessor.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.scripting.dsl;
+
+import java.lang.reflect.Constructor;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.io.Resource;
+import org.springframework.integration.handler.MessageProcessor;
+import org.springframework.integration.scripting.AbstractScriptExecutingMessageProcessor;
+import org.springframework.integration.scripting.RefreshableResourceScriptSource;
+import org.springframework.integration.scripting.ScriptVariableGenerator;
+import org.springframework.integration.scripting.jsr223.ScriptExecutingMessageProcessor;
+import org.springframework.integration.scripting.jsr223.ScriptExecutorFactory;
+import org.springframework.messaging.Message;
+import org.springframework.scripting.ScriptSource;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * The adapter {@link MessageProcessor} around {@link AbstractScriptExecutingMessageProcessor}.
+ * Delegates to the {@code GroovyScriptExecutingMessageProcessor}, if provided {@link #lang}
+ * matches to {@code groovy} string and {@code spring-integration-groovy} jar is in classpath.
+ * Otherwise to the {@link ScriptExecutingMessageProcessor}.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ */
+class DslScriptExecutingMessageProcessor
+		implements MessageProcessor<Object>, InitializingBean, ApplicationContextAware {
+
+	private static Constructor<?> groovyScriptExecutingMessageProcessorConstructor = null;
```

Comment:
This is a bit ugly; perhaps a cleaner implementation would be a subclass in the groovy module and register a factory bean in both places.